docs/aws/audit/elbmonitoring/rules/elb_no_deletion_policy.mdx

```diff
@@ -24,6 +24,208 @@ HIPAA, NIST
+   ```sh
+   aws elbv2 modify-load-balancer-attributes --load-balancer-arn <load-balancer-arn> --attributes Key=deletion_protection.enabled,Value=true
+   ```
+
+3. **Verify the Configuration:**
+   To ensure that the deletion protection flag has been enabled, describe the load balancer attributes using the following command:
+   ```sh
+   aws elbv2 describe-load-balancer-attributes --load-balancer-arn <load-balancer-arn>
+   ```
+   Look for the `deletion_protection.enabled` attribute in the output and ensure its value is `true`.
+
+4. **Automate for Multiple Load Balancers (Optional):**
+   If you have multiple load balancers and want to enable deletion protection for all of them, you can use a loop in a shell script. Here is an example:
+   ```sh
+   for lb_arn in $(aws elbv2 describe-load-balancers --query 'LoadBalancers[*].LoadBalancerArn' --output text); do
+       aws elbv2 modify-load-balancer-attributes --load-balancer-arn $lb_arn --attributes Key=deletion_protection.enabled,Value=true
+   done
+   ```
+
+By following these steps, you can ensure that deletion protection is enabled for your Elastic Load Balancers, preventing accidental deletions.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration where Elastic Load Balancers (ELBs) should have the Deletion Protection flag enabled in AWS, you can use the Boto3 library in Python. Here are the steps to achieve this:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+3. **Enable Deletion Protection for Existing ELBs**:
+   Use the following Python script to enable deletion protection for all existing ELBs in your AWS account.
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon EC2
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+
+   # Initialize the ELB client
+   elb_client = session.client('elb')
+
+   # Get a list of all load balancers
+   load_balancers = elb_client.describe_load_balancers()
+
+   # Iterate through each load balancer and enable deletion protection
+   for lb in load_balancers['LoadBalancerDescriptions']:
+       lb_name = lb['LoadBalancerName']
+       print(f"Enabling deletion protection for load balancer: {lb_name}")
+       elb_client.modify_load_balancer_attributes(
+           LoadBalancerName=lb_name,
+           LoadBalancerAttributes={
+               'DeletionProtection': {
+                   'Enabled': True
+               }
+           }
+       )
+       print(f"Deletion protection enabled for load balancer: {lb_name}")
+   ```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>